packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2071,6 +2025,10 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         } finally {
             this._orderSequentiallyCalls--;
         }
+
+        if (this.flushMode === FlushMode.Immediate && this._orderSequentiallyCalls === 0) {
```

Comment:
Isn't it !this.currentlyBatching() ?

Reply:
I initially had it as `!this.currentlyBatching()` but thought explicitly showing the condition is much clearer